tests/test_updown.py

```diff
@@ -1,147 +1,70 @@
-    import ctadata
-    ctadata.APIClient.downloadservice = start_service['url']
-
-    subprocess.check_call([
-        "dd", "if=/dev/random", "of=local-file-example", "bs=1M", "count=1000"
-    ])
-
-    r = ctadata.upload_file('local-file-example', 'example-files/example-file')
-    print(r)
-
-    ctadata.fetch_and_save_file(r['path'], 'restored-file-example')
-
-
-def test_apiclient_upload_invalid_path(start_service, caplog):
-    import ctadata
-    ctadata.APIClient.downloadservice = start_service['url']
-
-    subprocess.check_call(
-        ["dd", "if=/dev/random", "of=local-file-example", "bs=1M", "count=1"])
-
-    with pytest.raises(ctadata.api.StorageException):
-        ctadata.upload_file('local-file-example', '../example-file')
-
-
-def test_apiclient_upload_wrong(start_service, caplog):
-    import ctadata
-    ctadata.APIClient.downloadservice = start_service['url']
-
-    subprocess.check_call(
-        ["dd", "if=/dev/random", "of=local-file-example", "bs=1M", "count=1"])
-
-    with pytest.raises(ctadata.api.StorageException):
-        ctadata.upload_file('local-file-example',
-                            'example-files/example-file/../')
-
-
-def test_apiclient_upload_dir(start_service, caplog):
-    import ctadata
-    ctadata.APIClient.downloadservice = start_service['url']
-
-    with tempfile.TemporaryDirectory() as tmpdir:
-        for i in range(10):
-            subprocess.check_call([
-                "dd", "if=/dev/random", f"of={tmpdir}/local-file-example-{i}",
-                "bs=1M", "count=1"
-            ])
-
-        ctadata.upload_dir(tmpdir, 'example-files/tmpdir')
-
-
-@pytest.mark.xfail(reason="dav not implemented yet")
-def test_dav_list(start_service):
-    from webdav4.client import Client
-
-    client = Client(start_service['url'] + "/dav/lst")
-    client.exists("bla")
-
-    client.ls("", detail=True)
-    client.upload_file("test")
+import xmltodict
+import tempfile
+from conftest import webdav_server, generate_random_file, hash_file
+
+
+@pytest.mark.timeout(30)
+def test_health(app: Any, client: Any):
+    with webdav_server():
+        with app.app_context():
+            print(url_for('health'))
+            r = client.get(url_for('health'))
+            print(r.json, url_for('health'))
+            assert r.status_code == 200
+
+
+@pytest.mark.timeout(30)
+def test_list(app: Any, client: Any):
+    with webdav_server():
+        with app.app_context():
+            r = client.get(url_for('list', path="lst"))
+            assert r.status_code == 200
+            print(r.json)
+
+
+@pytest.mark.timeout(30)
+def test_download(app: Any, client: Any):
+    with webdav_server() as server:
+        filename = "md5sum-lst.txt"
+        remote_file = f"{server['config']['provider_mapping']['/']}/{filename}"
+        generate_random_file(remote_file, 1 * (1024**2))
+
+        with app.app_context():
+            r = client.get(url_for('fetch', path=filename))
+            assert r.status_code == 200
```

Comment:
non-webdav upload, which automatically selects directory according to the user name, should be tested here somewhere I guess